libs/azure-storage/README.md

```diff
@@ -1,9 +1,147 @@
 # langchain-azure-storage
 
-This package contains the LangChain integrations for Azure Storage. This project is a work-in-progress and currently does not contain any features.
+This package contains the LangChain integrations for Azure Storage. Currently, it includes:
+- `AzureBlobStorageLoader` - Loads LangChain document objects from Azure Blob Storage
 
 ## Installation
 
 ```bash
 pip install -U langchain-azure-storage
 ```
+
+## Document Loader Examples
+
+### Load from container
+Below shows how to load documents from all blobs in a given container in Azure Blob Storage:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+The example below shows how to load documents from blobs in a container with a given prefix:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+    prefix="test",
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+### Load from a list of blobs
+Below shows how to load documents from a list of blobs in Azure Blob Storage:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+    blob_names=["blob-1", "blob-2", "blob-3"],
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+### Load asynchronously
+Below shows how to load documents asynchronously which can be done by using either `aload()` or `alazy_load()`:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+async def main():
+    loader = AzureBlobStorageLoader(
+        account_url=<your_account_url>,
+        container_name=<your_container_name>,
+    )
+
+    async for doc in loader.alazy_load():
+        print(doc.page_content)
+```
```

Comment:
I think we can skip on having an example on async loading, `alazy_load()` is part of the base loader interface so customers will already know how that works most likely and really would be more interested on how to construct the loader.